lib/js-api/src/api/content-rest-api/api/settings.api.ts

```diff
@@ -0,0 +1,65 @@
+/*!
+ * @license
+ * Copyright © 2005-2023 Hyland Software, Inc. and its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { BaseApi } from './base.api';
+import { throwIfNotDefined } from '../../../assert';
+
+export class SettingsApi extends BaseApi {
+    /**
+     * Gets the published extension configuration from
+     * the database
+     *
+     * @param instanceId Unique ID for a running instance of ADW
+     * for which configuration is to be fetched
+     */
+    getSavedExtensionState(instanceId: string): Promise<any> {
+        throwIfNotDefined(instanceId, 'instanceId');
+
+        const pathParams = {
+            instanceId
+        };
+
+        return this.get({
+            path: '/settings/{instanceId}',
+            pathParams
+        });
+    }
+
+    /**
+     * Publish the extension configuration for an ADW instance
+     * in the database
+     *
+     * @param instanceId Unique ID for a running instance of ADW
```

Comment:
we should not mention ADW in description because js-api is reusable library so should know nothing about ADW